runtime/Go/antlr/tree.go

```diff
@@ -28,7 +28,7 @@ type SyntaxTree interface {
 type ParseTree interface {
 	SyntaxTree
 
-	Accept(Visitor ParseTreeVisitor) interface{}
+	Accept(visitor ParseTreeVisitor)
```

Comment:
My approach came out of creating an expression evaluator which had no mutable state of its own, which allows `.Accept(v)` to work fine.  If your visitor type has to carry changing state with it, then yes, you can't really call back into it as the state will get confused.  But you shouldn't HAVE to do that, and I didn't think your implementation forced that.

I also do this in a standard Java ANTLR-generated vistor, which I know shouldn't mean that's relevant for Go, but I mean it more as a point that it might be expected by ANTLR users.

If you imagine something which implements a simple expression parser:
```antlr
expr
    : '(' expr ')'                 #ParenExpr
    | a=expr op=(ADD|SUB) b=expr   #AddSubExpr
    | literal                      #LiteralExpr
    ;

literal
    : lit=INT                      #IntLiteral
    ;

SUB : '-' ;
ADD : '+' ;
INT :   [0-9]+ ;
```

An obvious implementation of `VisitIntLiteral(...)` is
```go
	result, err := strconv.ParseInt(ctx.GetText(), 10, 64)
	if err == nil {
		return result
	}
	return nil
```

That lets your `AddSubExpr` visitor do something like what I said at the top, and then (after a type check) add the numbers and return them.  Now you have a recursive visitor that evaluates expressions.  Essentially an interpreter, using the Visitor pattern to execute it.

(The one I'm working with in my current project is more complicated, but that's all extracted from pieces of it that are emblematic of the whole.  The interpreter and surrounding code is written in Java and Go, but both use the same grammar, and the same aspects of ANTLR.)